SmartDeviceLink/private/SDLStreamingVideoLifecycleManager.m

```diff
@@ -659,6 +690,242 @@ - (void)sdl_hmiStatusDidChange:(SDLRPCNotificationNotification *)notification {
+        self.preferredFormats = @[format];
+        self.preferredResolutions = @[resolution];
+        if (self.focusableItemManager != nil) {
+            self.focusableItemManager.enableHapticDataRequests = NO;
+        }
+        SDLLogD(@"Using generic video capabilites, preferred formats: %@, resolutions: %@, haptics disabled", self.preferredFormats, self.preferredResolutions);
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wnonnull"
+        [self sdl_useVideoCapability:nil];
+#pragma clang diagnostic pop
+    }
+}
+
+#pragma mark - Video capability logic
+
+- (void)sdl_useVideoCapability:(SDLVideoStreamingCapability *)videoCapability {
+    if ([self.videoStreamStateMachine.currentState isEqualToEnum:SDLVideoStreamManagerStateStarting]) {
+        [self sdl_applyVideoCapabilityWhileStarting:videoCapability];
+    } else if ([self.videoStreamStateMachine.currentState isEqualToEnum:SDLVideoStreamManagerStateReady]) {
+        [self sdl_applyVideoCapabilityWhenStreaming:videoCapability];
+    } else {
+        SDLLogD(@"Received video capabilities in wrong state: %@; ignore it", self.videoStreamStateMachine.currentState);
+    }
+}
+
+- (void)sdl_applyVideoCapabilityWhileStarting:(nullable SDLVideoStreamingCapability *)videoCapabilityUpdated {
+    SDLVideoStreamingCapability *videoCapability = (videoCapabilityUpdated == nil) ? [self sdl_defaultVideoCapability] : videoCapabilityUpdated;
+    NSArray<SDLVideoStreamingCapability *> *capabilityMatches = [self matchVideoCapability:videoCapability];
+    [self sdl_sendOnAppCapabilityUpdated:capabilityMatches];
+
+    if (capabilityMatches.count == 0) {
+        // use default video capability if no match found
+        capabilityMatches = (videoCapabilityUpdated == nil) ? @[[self sdl_defaultVideoCapability]] : @[[videoCapabilityUpdated shortCopy]];
+    }
+
+    SDLVideoStreamingCapability *matchedVideoCapability = capabilityMatches.firstObject;
+    if (capabilityMatches.count > 1) {
+        // add the remaining capabilities that match video streaming criteria
+        matchedVideoCapability.additionalVideoStreamingCapabilities = [capabilityMatches subarrayWithRange:NSMakeRange(1, capabilityMatches.count - 1)];
+    } else {
+        // add any other possible capabilities here if needed
+        matchedVideoCapability.additionalVideoStreamingCapabilities = @[];
+    }
+
+    // take formats from 'parent' since onse may be absent in children
+    if (!matchedVideoCapability.supportedFormats) {
+        matchedVideoCapability.supportedFormats = videoCapabilityUpdated.supportedFormats;
+    }
+    [self sdl_applyVideoCapability:matchedVideoCapability];
+
+    if (self.delegate) {
+        __weak typeof(self) weakSelf = self;
+        const CGSize displaySize = matchedVideoCapability.makeImageResolution.makeSize;
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [weakSelf.delegate videoStreamingSizeDidUpdate:displaySize];
+        });
+    }
```

Comment:
I wonder if notifying the observer should happen after we get the video start service ACK? Otherwise we are telling them to change the display size before we even know if the module will allow video streaming.

Reply:
The delegate must know if the size updated. It is up to the client app to take an action or not based on the call.